Add output folder icons

Michoko92 opened the pull request:
Adds icons on the first 3 tabs to directly open the corresponding images output directory.

Michoko92 committed:
Add output folder icons

Adds icons on the first 3 tabs to directly open the corresponding images output directory

dfaker commented:
A the very least a good way to clutter a server with thousands of open windows!

I'd suggest at least an option to allow people serving their friends an instance from their pc to choose to turn it off.

Things may have changed but I don't think this is cross platform, you may need to put switches in for some OS variations:

```python
if platform.system() == "Windows":
    os.startfile(path)
elif platform.system() == "Darwin":
    sp.Popen(["open", path])
else:
    sp.Popen(["xdg-open", path])
```

Has never raised bugs for me, but the world is a big place.

I'd suspect this will be declined as it's crossing the web barrier into the local system, but we'll see.

AUTOMATIC1111 commented:
I was actually willing to accept until your message dfaker.

So the `--hide-ui-dir-config` flag should be honored and the button should be hidden if it's on, `open_folder` should do nothing if the flag is on, and this has to be tested on linux.

Michoko92 committed:
Add output folder icons

Handling of the --hide-ui-dir-config flag and added multi-platform code for opening a folder

Michoko92 commented:
OK now if --hide-ui-dir-config is used, the icons are hidden and the open_folder function won't do anything.

Unfortunately, I'm not able to test this on Linux, but I suppose dfaker's code is battle-tested?

Michoko92 committed:
Add output folder icons

Changed the hidden element class name to a more generic one, so people can reuse it if they want to hide further elements in the future

blanelegant commented:
This button does nothing when being hosted on another machine on the local network or internet. It should be hidden in this case.

Michoko92 commented:
If it is hosted, shouldn't the UI be launched with the --hide-ui-dir-config argument anyway? If this flag is put in the command line, it hides the folders configuration in the settings, as well as the folder icons.

leszekhanusz commented:
FYI this PR broke the `/config` path. I made the PR #1532 to fix it.

Michoko92 commented:
Thank you! 👍
